addons/web/static/src/views/graph/graph_renderer.js

```diff
@@ -360,24 +482,21 @@ export class GraphRenderer extends Component {
      * @returns {Object}
      */
     getLineChartData() {
-        const { groupBy, domains, stacked, cumulated } = this.model.metaData;
+        const { cumulated, groupBy, domains } = this.model.metaData;
         const data = this.model.data;
-        const color0 = getColor(0, cookie.get("color_scheme"));
-        const color1 = getColor(1, cookie.get("color_scheme"));
         for (let index = 0; index < data.datasets.length; ++index) {
             const dataset = data.datasets[index];
+            const itemColor =  getColor(index, colorScheme, data.datasets.length);
+            dataset.backgroundColor = getCustomColor(colorScheme, lightenColor(itemColor, .5), darkenColor(itemColor, .5));
+            dataset.cubicInterpolationMode = 'monotone';
+            dataset.borderColor = itemColor;
+            dataset.borderWidth = 2;
+            dataset.hoverBackgroundColor = dataset.borderColor;
+            dataset.pointRadius = 3;
+            dataset.pointHoverRadius = 6;
+
             if (groupBy.length <= 1 && domains.length > 1) {
-                if (dataset.originIndex === 0) {
```

Comment:
It seems originIndex does not play a big role now. Is it normal (we have multiple originIndexes in comparison mode (date filter active + comparsion active (e.g.  Previous period) ))

Reply:
As the colors are now flat and well defined according to the number of elements displayed [here](https://github.com/odoo/odoo/pull/167296/files#diff-35a97df8abfcb7b324f37344af4b928f54cad31a67086fa3a35c569a3d8fc74aR489-R490), I think this is no longer necessary. Or perhaps I've missed something? 🤔

Reply:
I think this code was here to handle a special case. When only one dataset is available for each domain, the the curve for the main origin was filled under and the other one not. This was a way to make evident what is the data from the main period

Reply:
I've made some tests on `master` to trigger this condition, but if I'm not mistaken, the result is the same as the default behavior. 🤔 
<img width="1789" alt="Capture d’écran 2024-06-25 à 15 02 25" src="https://github.com/odoo/odoo/assets/80679690/238f74ea-9519-40ec-bc76-6075bf7cea7b">

Reply:
You should try with Creation Date: day as groupby

Reply:
As we've discussed, we'll keep it as it is and modify it later if necessary.